src/rdiffbackup/utils/psutil.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 Eric Lavarde
+#
+# This file is part of rdiff-backup.
+#
+# rdiff-backup is free software; you can redistribute it and/or modify
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# rdiff-backup is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with rdiff-backup; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA
+"""
+The well known psutil library isn't really working the way I expect,
+especially not under Windows, hence our own implementation is used.
+
+It's perhaps not the most efficient approach but it doesn't need to be as
+it's called only seldomly.
+"""
+import os
+import subprocess
+
+
+def get_pid_name(pid):
+    """
+    Gets a Process ID as integer or string and returns a process/image name
+    Returns None if no process is running with the given PID
+    """
+    if os.name == "nt":
+        cmd = ("tasklist", "/nh", "/fi", "pid eq {pp}".format(pp=pid))
+    else:
+        cmd = ("ps", "q", str(pid), "o", "comm=", "o", "pid=")
```

Comment:
Using `ps` might not be the most stable approach. To my knowledge, the arguments are not stable between Linux and Unix. 
On Debian:
```
$ ps q 1493225 o comm= o pid=
gnome-screensho 1493225
```

On MacOS:
```
% ps q 37511 o comm= o pid=
ps: illegal option -- q
usage: ps [-AaCcEefhjlMmrSTvwXx] [-O fmt | -o fmt] [-G gid[,gid...]]
          [-u]
          [-p pid[,pid...]] [-t tty[,tty...]] [-U user[,user...]]
       ps [-L]
```

I'm a bit surprise psutil is not covering our need. On Windows I get this:
![image](https://user-images.githubusercontent.com/921251/163569039-7408f9c8-8361-4485-9838-9436e8a79292.png)

On Linux Debian I get this: 
![image](https://user-images.githubusercontent.com/921251/163569151-0a77d938-c3b7-49fb-bcc2-20f7416feaa4.png)

On MacOS
![image](https://user-images.githubusercontent.com/921251/163569545-8771729d-6a0a-4712-9b7f-1b180c9c5011.png)

Reply:
Under WIndows, psutil was still listing processes which were not running anymore, they were still listed but had a state `stopped` (or something like this). I can't reproduce it right now (of course), and we should indeed keep MacOS running. I assumed it would follow the BSD format... OK, let me look into this once again.